### PR TITLE
Fix for excessive memory usage on mailboxes with thousands of folders

### PR DESCRIPTION
**Pull request by leecher1337**
https://bugs.sogo.nu/view.php?id=5884

### CODE REVIEW

File: ActiveSync/SOGoActiveSyncDispatcher+Sync.m [hunk 1]
```diff
@@ -2627,10 +2627,10 @@ - (void) processSync: (id <DOMElement>) theDocumentElement
       // We first check of any of the collections we want to sync are already
       // in an other sync process. If that's the case, we do not do anything
       // and we return immediately. So we'll let the other sync process terminate
+      globalMetadata = [self globalMetadataForDevice];
```
Comment (WoodySlum) [inline ActiveSync/SOGoActiveSyncDispatcher+Sync.m:2630]: Can you also use this variable L2640 ?

**Review by WoodySlum — CHANGES_REQUESTED** [review]
Thank you for contributing to SOGo. It seems good to me except for the commit naming. I will fix it when merging.
Can you also replace it line https://github.com/Alinto/sogo/blob/e748cdc8fbde382917d2991663a8390d7c35ceae/ActiveSync/SOGoActiveSyncDispatcher%2BSync.m#L2640 ?
@tfux as you know EAS part well is it ok for you also ?

File: ActiveSync/SOGoActiveSyncDispatcher+Sync.m (hunk 1 above)
Comment (leecher1337) [reply ActiveSync/SOGoActiveSyncDispatcher+Sync.m:2630]: Is it OK now?

**Comment by WoodySlum** [comment]
Yes thank you very much. I'm just waiting for a feedback of @tfux, but the
fix seems for me



Le mar. 17 oct. 2023 à 09:35, leecher1337 ***@***.***> a
écrit :

File: ActiveSync/SOGoActiveSyncDispatcher+Sync.m (hunk 1 above)
Comment (WoodySlum) [reply ActiveSync/SOGoActiveSyncDispatcher+Sync.m:2630]: Nice thank you

**Review by WoodySlum — APPROVED** [review]

**Comment by tfux** [comment]
Fix is ok. Good catch.

**Comment by WoodySlum** [comment]
Thank you all !

**Pull request merged**